core/base/inc/RtypesCore.h

```diff
@@ -85,7 +85,8 @@ typedef float          Size_t;      //Attribute size (float)
 //---- constants ---------------------------------------------------------------
 
 #ifndef NULL
-#define NULL 0
+constexpr const auto R__DEPRECATED(6,12, "Please #include <cstddef> instead of relying on ROOT's NULL")
+   NULL = nullptr;
```

Comment:
I believe it would be better to either use `#define NULL ((void*)0)` to be compatible with C, or simply remove it from ROOT altogether and just `#include <cstddef>` to get it from the standard. I wonder what happens when one uses `nullptr` to call something from a C library. Won't things break? If not, then just including the standard header is probably the best solution. References: [C](http://en.cppreference.com/w/c/types/NULL) and [C++](http://en.cppreference.com/w/cpp/types/NULL) versions of NULL.